### PR TITLE
Allow passing jobs arguments to make command in qmk cli (qmk#10918)

### PR DESCRIPTION
**Pull request by ReVanTis**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [X] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [X] Documentation

## Issues Fixed or Closed by This PR

#10918

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [X] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by ReVanTis** [comment]
I'm a bit confused by the status of CLI CI test.
It shows OK in the result but labels as failing.
Seems the return code is somehow wrong in this case.

```
----------------------------------------------------------------------
Ran 36 tests in 24.783s

OK
Error: Process completed with exit code 1.
```

File: lib/python/qmk/commands.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 import qmk.keymap
 
 
-def create_make_command(keyboard, keymap, target=None):
+def create_make_command(keyboard, keymap, target=None, jobs = 1):
```
Comment (fauxpark) [inline lib/python/qmk/commands.py:13]:
```suggestion
def create_make_command(keyboard, keymap, target=None, jobs=1):
```

File: lib/python/qmk/commands.py [hunk 2]
```diff
@@ -23,6 +23,9 @@ def create_make_command(keyboard, keymap, target=None):
 
         target
             Usually a bootloader.
+
```
Comment (fauxpark) [inline lib/python/qmk/commands.py:26]:
```suggestion

```

File: lib/python/qmk/commands.py [hunk 3]
```diff
@@ -23,6 +23,9 @@ def create_make_command(keyboard, keymap, target=None):
 
         target
             Usually a bootloader.
+        
+        jobs
+            Specify number of parallel jobs for compiling. Default 1.
```
Comment (fauxpark) [inline lib/python/qmk/commands.py:28]:
```suggestion
            Specify number of parallel jobs for compiling. Default: 1.
```

File: lib/python/qmk/tests/test_cli_commands.py [hunk 4]
```diff
@@ -28,6 +28,9 @@ def test_compile():
     result = check_subcommand('compile', '-kb', 'handwired/onekey/pytest', '-km', 'default', '-n')
     check_returncode(result)
 
+def test_compile_with_jobs():
+    result = check_subcommand('compile', '-kb', 'handwired/onekey/pytest', '-km', 'default', '-j', '4', '-n')
+    check_returncode(result)
```
Comment (fauxpark) [inline lib/python/qmk/tests/test_cli_commands.py:33]:
```suggestion

def test_compile_with_jobs():
    result = check_subcommand('compile', '-kb', 'handwired/onekey/pytest', '-km', 'default', '-j', '4', '-n')
    check_returncode(result)

```

File: lib/python/qmk/tests/test_cli_commands.py [hunk 5]
```diff
@@ -38,6 +41,9 @@ def test_flash():
     result = check_subcommand('flash', '-kb', 'handwired/onekey/pytest', '-km', 'default', '-n')
     check_returncode(result)
 
+def test_flash_with_jobs():
+    result = check_subcommand('flash', '-kb', 'handwired/onekey/pytest', '-km', 'default', '-j', '4', '-n')
+    check_returncode(result)
```
Comment (fauxpark) [inline lib/python/qmk/tests/test_cli_commands.py:46]:
```suggestion

def test_flash_with_jobs():
    result = check_subcommand('flash', '-kb', 'handwired/onekey/pytest', '-km', 'default', '-j', '4', '-n')
    check_returncode(result)

```

**Review by fauxpark — CHANGES_REQUESTED** [review]
yapf is complaining that the code is not formatted correctly.

```
lib/python/qmk/commands.py:13:60: E251 unexpected spaces around keyword / parameter equals
lib/python/qmk/commands.py:13:62: E251 unexpected spaces around keyword / parameter equals
lib/python/qmk/commands.py:26:1: W293 blank line contains whitespace
lib/python/qmk/tests/test_cli_commands.py:31:1: E302 expected 2 blank lines, found 1
lib/python/qmk/tests/test_cli_commands.py:35:1: E302 expected 2 blank lines, found 1
lib/python/qmk/tests/test_cli_commands.py:44:1: E302 expected 2 blank lines, found 1
lib/python/qmk/tests/test_cli_commands.py:48:1: E302 expected 2 blank lines, found 1
```

**Comment by ReVanTis** [comment]
Hi @fauxpark,
Thanks for your review. I've applied your requested change for formats. Please kindly have a look!

**Comment by zvecr** [comment]
Does this add value over just using `MAKEFLAGS`? That way we don't have to implement anything, apart from maybe docs.

**Comment by ReVanTis** [comment]
> Does this add value over just using `MAKEFLAGS`? That way we don't have to implement anything, apart from maybe docs.

Actually, this extends the _-jX_ arguments to the real make command, which, is substantially the same to modifying MAKEFLAGS in this simple case.
And, of course, If QMK CLI prefers to let user pick MAKEFLAGS themselves, that could work too.

For further discussion, maybe we can extend this PR to pass _any_ arguments to the _make_ command for more flexibility.

**Comment by zvecr** [comment]
Locally I have `MAKEFLAGS` exported in my direnv config, however something like the following also works fine,
```
MAKEFLAGS="-j4" qmk compile -kb lets_split -km default
```

> maybe we can extend this PR to pass any arguments to the make command for more flexibility

Given my previous example supports this, why add more code that we have to support when we don't have to? I will leave the final call down to @skullydazed and @Erovia as this is more their domain.

**Comment by skullydazed** [comment]
I've been thinking about think one and I think I agree with @zvecr. Since it can be done with `MAKEFLAGS` and it's not a behavior we necessarily want to promote I'd rather not bake support into `qmk compile`. Some documentation explaining `MAKEFLAGS` that uses `-j` as an example would be a good way to go.

**Comment by ReVanTis** [comment]
Sorry that I was just too lazy to typing 'MAKEFLAGS="-jX"' for each time I update my firmware.
For me, modifying it globally is also not an good idea because I'm also maintaining some ancient code that went really bad if I'm not compiling with more than one jobs. So I thought it could save a few key types if I pass the "-jX" argument to the command.
But anyway, I agree it would be tidier to keep it this way.
I'm closing this one now and BIG thanks for everyone!

**Comment by zvecr** [comment]
@ReVanTis have a look into direnv, the workflow is quite good when you don't want to modify globally (which is the same stance I have). Along with a few other things I have, it's "cd into qmk_firmware folder, MAKEFLAGS magically set".

**Pull request closed without merging**